src/java/org/apache/cassandra/repair/autorepair/FixedSplitTokenRangeSplitter.java

```diff
@@ -33,68 +34,49 @@
 
 import static org.apache.cassandra.repair.autorepair.AutoRepairUtils.split;
 
-public class DefaultAutoRepairTokenSplitter implements IAutoRepairTokenRangeSplitter
+public class FixedSplitTokenRangeSplitter implements IAutoRepairTokenRangeSplitter
 {
-    @Override
-    public Iterator<KeyspaceRepairAssignments> getRepairAssignments(AutoRepairConfig.RepairType repairType, boolean primaryRangeOnly, List<PrioritizedRepairPlan> repairPlans)
+    /**
+     * The number of subranges to split each to-be-repaired token range into,
+     * the higher this number, the smaller the repair sessions will be
+     * How many subranges to divide one range into? The default is 1.
+     * If you are using v-node, say 256, then the repair will always go one v-node range at a time, this parameter, additionally, will let us further subdivide a given v-node range into sub-ranges.
+     * With the value “1” and v-nodes of 256, a given table on a node will undergo the repair 256 times. But with a value “2,” the same table on a node will undergo a repair 512 times because every v-node range will be further divided by two.
+     * If you do not use v-nodes or the number of v-nodes is pretty small, say 8, setting this value to a higher number, say 16, will be useful to repair on a smaller range, and the chance of succeeding is higher.
+     */
```

Comment:
Small suggestion for better readability. Also, this documentation feels like it belongs in cassandra.yaml and not hidden in the code.

```suggestion
        /**
     * The number of subranges to split each to-be-repaired token range into.
     * The higher this number, the smaller the repair sessions will be
     * How many subranges to divide one range into? The default is 1.
     * If you are using v-nodes, say 256, then the repair will always go one
     * v-node range at a time, this parameter, additionally, will let us
     * further subdivide a given v-node range into sub-ranges.
     * With the value "1" and v-nodes of 256, a given table on a node will
     * undergo the repair 256 times. But with a value "2", the same table
     * on a node will undergo a repair 512 times because every v-node range
     * will be further divided by two.
     * If you do not use v-nodes or the number of v-nodes is pretty small,
     * say 8, setting this value to a higher number, say 16, will be useful
     * to repair on a smaller range, and the chance of succeeding is higher.
     */
```

Reply:
Agreed, I ported this over from `AutoRepairConfig` to the splitter itself as is.   I think we need to make dedicated docs on this and also notably update cassandra.yaml as you mention.

I think we need to take a pass at this generally, as you can see in `RepairTokenRangeSplitter` I added a bunch of documentation around defaults, and `AutoRepairConfig` itself has a bunch.  Will implement your suggestion, but mind if we tackle this as another PR?